src/mcfret.c

```diff
@@ -0,0 +1,649 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <math.h>
+#include <string.h>
+#include <time.h>
+#include "omp.h"
+#include "types.h"
+#include "NISE_subs.h"
+#include "mcfret.h"
+#include "project.h"
+
+/* Main MCFRET routine only calling and combining the other subroutines */ 
+void mcfret(t_non *non){
+    int nn2;
+    int segments;
+  /* Response functions for emission and absorption: real and imaginary part*/
+    float *re_Abs,*im_Abs;
+    float *re_Emi,*im_Emi;
+    float *J;
+    float *rate_matrix;
+
+  /*Allocate memory for the response functions*/
+    nn2=non->singles*non->singles;
+    re_Abs=(float *)calloc(nn2*non->tmax,sizeof(float));
+    im_Abs=(float *)calloc(nn2*non->tmax,sizeof(float));
+    re_Emi=(float *)calloc(nn2*non->tmax,sizeof(float));
+    im_Emi=(float *)calloc(nn2*non->tmax,sizeof(float));
+    J=(float *)calloc(nn2,sizeof(float));
+
+  /* The rate matrix is determined by the integral over t1 for */
+  /* Tr [ J * Abs(t1) * J * Emi(t1) ] */
+
+    segments=project_dim(non);
+    if (segments<2){
+      printf(RED "Too few segments defined for MCFRET calculation!" RESET);
+      exit(0);
+    }
+    rate_matrix=(float *)calloc(segments*segments,sizeof(float));
+
+  /* Tell the user that we are in the MCFRET Routine */
+     if (!strcmp(non->technique, "MCFRET") || (!strcmp(non->technique, "MCFRET-Autodetect")) || (!strcmp(non->technique, "MCFRET-Absorption"))
+      || (!strcmp(non->technique, "MCFRET-Emission")) || (!strcmp(non->technique, "MCFRET-Coupling")) || (!strcmp(non->technique,
+      "MCFRET-Rate")) || (!strcmp(non->technique, "MCFRET-Analyse")) ) {
+          printf("Perfroming the MCFRET calculation.\n");
+    }
+
+/* Call the absorption routine */
+    if (!strcmp(non->technique, "MCFRET") || (!strcmp(non->technique, "MCFRET-Absorption"))){
```

Comment:
Calling the technique "MCFRET-Absorption" or "MCFRET-Emission" do not result in any output.